doc/read-the-docs-site/doc/marconi-as-a-library/tutorials/BasicApp.hs

```diff
@@ -0,0 +1,484 @@
+
+-- BLOCKEND env
+
+-- BLOCKSTART runBlockInfoListIndexer
+runBlockInfoListIndexer :: IO ()
+runBlockInfoListIndexer = do
+  worker <- mkBlockInfoListIndexerWorker
+  withIndexerBuildEnv worker $ \env -> liftIO $ runIndexer env
+
+-- BLOCKEND runBlockInfoListIndexer
+
+-- BLOCKSTART runBlockInfoSqliteIndexer
+runBlockInfoSqliteIndexer :: IO ()
+runBlockInfoSqliteIndexer = do
+  worker <- mkBlockInfoSqliteIndexerWorker ":memory:"
+  withIndexerBuildEnv worker $ \env -> liftIO $ runIndexer env
+
+-- BLOCKEND runBlockInfoSqliteIndexer
+
+-- BLOCKSTART runBlockInfoIndexer
+runIndexer :: Env indexer -> IO ()
+runIndexer env = do
+  -- TODO Why do we need a coordinator here? We should be able to start a worker with `runIndexer`.
+  indexer <-
+    liftIO $ Core.mkCoordinator [Core.worker $ envIndexerWorker env]
+  liftIO $
+    Core.runIndexer
+      ( Core.RunIndexerConfig
+          -- The 'Trace' used for logging
+          (envTrace env)
+          -- The block preprocessor. Will be explained in another tutorial.
+          Core.withNoPreprocessor
+          -- The retry configuration. If can't connect to local node, or lost
+          -- connection, we retry given this configuration.
+          (Core.RetryConfig 30 Nothing)
+          -- The security parameter
+          (envSecurityParam env)
+          -- The NetworkId
+          (envNetworkId env)
+          -- The point from which we want to start the chain-sync protocol.
+          C.ChainPointAtGenesis
+          -- UNIX socket file path exposed by the local node.
+          (envSocketFilePath env)
+      )
+      -- The indexer to run given the previous configuration.
+      indexer
+
+-- BLOCKEND runBlockInfoIndexer
+
+-- BLOCKSTART runBlockInfoListIndexerHttp
+runBlockInfoListIndexerHttp :: IO ()
+runBlockInfoListIndexerHttp = do
+  worker <- mkBlockInfoListIndexerWorker
+  withIndexerBuildEnv worker $ \env -> do
+    let blockInfoWorker = envIndexerWorker env
+    let blockInfoIndexerVar = Core.workerIndexerVar blockInfoWorker
+
+    -- Run two threads: the indexing thread and query thread
+    -- They each share an `MVar indexer`.
+    race_
+      (runHttpServer blockInfoIndexerVar)
+      (runIndexer env)
+
+-- BLOCKEND runBlockInfoListIndexerHttp
+
+-- BLOCKSTART runBlockInfoSqliteIndexerHttp
+runBlockInfoSqliteIndexerHttp :: IO ()
+runBlockInfoSqliteIndexerHttp = do
+  worker <- mkBlockInfoSqliteIndexerWorker ":memory:"
+  withIndexerBuildEnv worker $ \env -> do
+    let blockInfoWorker = envIndexerWorker env
+    let blockInfoIndexerVar = Core.workerIndexerVar blockInfoWorker
+
+    -- Run two threads: the indexing thread and query thread
+    -- They each share an `MVar indexer`.
+    race_
+      (runHttpServer blockInfoIndexerVar)
+      (runIndexer env)
+
+-- BLOCKEND runBlockInfoSqliteIndexerHttp
+
+-- BLOCKSTART event
+
+-- In Marconi, each block received by the local node is sent to all Marconi
+-- indexers. Each indexer needs to define how a block translates to an event.
```

Comment:
I think this is an important part of the logic that Marconi implements behind the scenes. It would be helpful to talk a bit about this more, and link to relevant code. This might be out of the scope of this specific tutorial, but it might be worth elaborating in a separate tutorial about `marconi-core` and linking to that here.

Reply:
That's right, it's `marconi-chain-index`, but it will be moved to `marconi-core-cardano` eventually.

Reply:
Add more details. Can you revalidate? https://github.com/input-output-hk/marconi/pull/202/files#diff-e92ed9cee78c396f00fdc33f17b96888edd487292f32bb3ebd9bcee50062eb3fR187